### PR TITLE
Added const to interface

### PR DESCRIPTION
**Pull request by kpp**
don't touch it. in progress

### CODE REVIEW

**Comment by heinermann** [comment]
I don't see any benefit to making the arguments const besides correctness, for some definition of correctness. In some cases when things under the hood change you would need to change their constness back to non-const, which could break the API. Some functions which could be const are not because they simply write to the structure (i.e. any order and the lastOrder member). I don't want to use const_cast anywhere either. What are your thoughts?

**Comment by kpp** [comment]
@heinermann I wrote a 20kb comment, but I burnt it because of rage. If something is not changing it must be `const`.

File: build.bat [hunk 1]
```diff
@@ -22,9 +22,9 @@ if defined APPVEYOR (
 :: Build BWAPI's full stack
 cd bwapi
 ::nuget restore
-msbuild %MSBUILD_ADDITIONAL_OPTIONS% /verbosity:normal /p:Configuration=Debug_Pipeline bwapi.sln
-msbuild %MSBUILD_ADDITIONAL_OPTIONS% /verbosity:normal /p:Configuration=Release_Pipeline bwapi.sln
-msbuild %MSBUILD_ADDITIONAL_OPTIONS% /verbosity:normal /p:Configuration=Installer_Target bwapi.sln
+msbuild %MSBUILD_ADDITIONAL_OPTIONS% /verbosity:normal /p:Configuration=Debug_Pipeline bwapi.sln || exit 1
+msbuild %MSBUILD_ADDITIONAL_OPTIONS% /warnaserror /verbosity:normal /p:Configuration=Release_Pipeline bwapi.sln || exit 1
```
Comment (heinermann) [inline build.bat]: If you're setting warnaserror here you might want to set the same flag in the VS project files.

File: bwapi/include/BWAPI/Player.h [hunk 2]
```diff
@@ -643,6 +643,6 @@ namespace BWAPI
     /// @returns true if the unit type requirements are met, and false otherwise.
     ///
     /// @since 4.1.2
-    bool hasUnitTypeRequirement(UnitType unit, int amount = 1) const;
+    bool hasUnitTypeRequirement(const UnitType unit, const int amount = 1) const;
```
Comment (heinermann) [inline bwapi/include/BWAPI/Player.h]: @kpp Value types don't need to be marked const. It's verbose. (Note that the Type class and everything that extends it is a value type)

**Pull request closed without merging**